src/index.js

```diff
@@ -0,0 +1,51 @@
+import LogData from './log'
+
+function attachListener(){
+  window.addEventListener('mezzurite/componentsChanged', (e) => { 
+    MezzuritePlugin.log(MezzuritePlugin, e.detail); 
+  } , false);
+}
+
+function setNextPlugin (next) {
+  if (next) {
+    MezzuritePlugin._nextPlugin = next;
+  }
+}
+
+function processTelemetry (item) {
+    // use the generated page id for parent Id and use props.id as operation Id
+  if (item.tags['ai.operation.id']) {
+    let parentId = item.tags['ai.operation.id'];
+    let origId = item.baseData && item.baseData.properties && item.baseData.properties.id ? item.baseData.properties.id : undefined;
+    if (origId) {
+      item.tags['ai.operation.id'] = origId;
+      delete item.baseData.properties.id;
+      item.tags['ai.operation.parentId'] = parentId;
+    }
+  }
+  if (MezzuritePlugin._nextPlugin && MezzuritePlugin._nextPlugin.processTelemetry) {
+    MezzuritePlugin._nextPlugin.processTelemetry(item);
+  }
+}
+
+function initialize (config, core, extensions) {
+  extensions.forEach((e => {
```

Comment:
The `Array.prototype.find` function is better suited for what you're looking for here.